incubator-tools/best-practices/utilities/utilities.py

```diff
@@ -0,0 +1,743 @@
+    # adding entities which are present in file2 but not in file1
+    for row in df_file2.values:
+        df_compare.loc[len(df_compare.index)] = [
+            row[0],
+            "Entity not found.",
+            row[1],
+            "[]",
+            row[2],
+            "[]",
+            row[3],
+        ]
+
+    # df_compare['Match'] = df_compare['Ground Truth Text'] == df_compare['Output Text']
+    match_array = []
+    for i in range(0, len(df_compare)):
+        match_string = ""
+        if (
+            df_compare.iloc[i]["Pre_HITL_Output"] == "Entity not found."
+            and df_compare.iloc[i]["Post_HITL_Output"] == "Entity not found."
+        ):
+            match_string = "TN"
+        elif (
+            df_compare.iloc[i]["Pre_HITL_Output"] != "Entity not found."
+            and df_compare.iloc[i]["Post_HITL_Output"] == "Entity not found."
+        ):
+            match_string = "FN"
+        elif (
+            df_compare.iloc[i]["Pre_HITL_Output"] == "Entity not found."
+            and df_compare.iloc[i]["Post_HITL_Output"] != "Entity not found."
+        ):
+            match_string = "FP"
+        elif (
+            df_compare.iloc[i]["Pre_HITL_Output"] != "Entity not found."
+            and df_compare.iloc[i]["Post_HITL_Output"] != "Entity not found."
+        ):
+            if (
+                df_compare.iloc[i]["Pre_HITL_Output"]
+                == df_compare.iloc[i]["Post_HITL_Output"]
+            ):
+                match_string = "TP"
+            else:
+                match_string = "FP"
+        else:
+            match_string = "Something went Wrong."
+
+        match_array.append(match_string)
+
+    df_compare["Match"] = match_array
+
+    df_compare["Fuzzy Ratio"] = df_compare.apply(GetMatchRatio, axis=1)
+    if list(df_compare.index):
+        score = df_compare["Fuzzy Ratio"].sum() / len(df_compare.index)
+    else:
+        score = 0
+
+    # display(df_compare)
+    return df_compare, score
+
+
+def get_document_schema(location, project_number, processor_id, processor_version_id):
+    """
+    Fetches the document schema of a specific processor version from Google Cloud DocumentAI.
+
+    Args:
+        location (str): The location of the DocumentAI service ("eu" or other values).
+        project_number (str): The number representing the Google Cloud project.
+        processor_id (str): The ID of the DocumentAI processor.
+        processor_version_id (str): The ID of the processor version.
+
+    Returns:
+        documentai.types.Document.Schema: The document schema for the specified processor version.
+
+    Example:
+        schema = get_document_schema("eu", "123456", "processor123", "version123")
+    """
+    # Choose the endpoint based on the provided location.
+    opts = {}
+    if location == "eu":
+        opts = {"api_endpoint": "eu-documentai.googleapis.com"}
+    else:
+        opts = {"api_endpoint": "us-documentai.googleapis.com"}
```

Comment:
```suggestion
    # You must set the `api_endpoint` if you use a location other than "us".
    opts = ClientOptions(api_endpoint=f"{location}-documentai.googleapis.com")
```

Note: You'll also need to add this at the top of the file:

```py
from google.api_core.client_options import ClientOptions
```